protocol/x/clob/keeper/process_proposer_matches_events.go

```diff
@@ -40,6 +41,10 @@ func (k Keeper) MustSetProcessProposerMatchesEvents(
 			processProposerMatchesEvents.BlockHeight, ctx.BlockHeight()))
 	}
 
+	if err := processProposerMatchesEvents.ValidateProcessProposerMatchesEvents(); err != nil {
```

Comment:
Should we move the block height validation on lines 39-42 within this function as well?